tests/foreman/ui/test_computeresource.py

```diff
@@ -18,173 +21,197 @@
 class ComputeResource(UITestCase):
     """Implements Compute Resource tests in UI"""
 
-    @data(*generate_strings_list(len1=8))
-    def test_create_resource_1(self, name):
-        """@Test: Create a new libvirt Compute Resource
+    current_libvirt_url = (LIBVERT_RESOURCE_URL
+                           % conf.properties['main.server.hostname'])
+    current_docker_url = (DOCKER_RESOURCE_URL
+                          % conf.properties['main.server.hostname'])
+
+    @data(*valid_data_list())
+    def test_create_libvirt_resource_different_names(self, name):
+        """@Test: Create a new libvirt Compute Resource using different value
+        types as a name
 
         @Feature: Compute Resource - Create
 
-        @Assert: A libvirt Compute Resource is created
+        @Assert: A libvirt Compute Resource is created successfully
 
         """
-        libvirt_url = "qemu+tcp://%s:16509/system"
-        provider_type = FOREMAN_PROVIDERS['libvirt']
-        url = (libvirt_url % conf.properties['main.server.hostname'])
         with Session(self.browser) as session:
```

Comment:
Probably, better ask here:
https://github.com/SatelliteQE/robottelo/issues/2490

From my point of view, if we get rid from one redundant ```waiter_for_element```  call we can save more time than thousands of assignment operations will take, but I agree that we have to make our code as more optimal as we can.
And, actually, for current test, I move these assignment operation in one, single place